miso-web/src/main/webapp/scripts/list_printer.js

```diff
@@ -35,15 +35,78 @@ ListTarget.printer = {
       "name": "Enable",
       "action": function(items) {
         Utils.ajaxWithDialog('Enabling Printer', 'PUT', Urls.rest.printers.enable, items.map(Utils.array.getId), Utils.page.pageReload);
-
       }
     }, {
       "name": "Disable",
       "action": function(items) {
         Utils.ajaxWithDialog('Disabling Printer', 'PUT', Urls.rest.printers.disable, items.map(Utils.array.getId), Utils.page.pageReload);
 
       }
-    }, ListUtils.createBulkDeleteAction("Printers", 'printers', Utils.array.getName)];
+    }, {
+      "name": "Duplicate",
+      "action": function(items) {
+        if (items.length > 1) {
+          Utils.showOkDialog('Duplicate Printer', ['One at a time please.']);
+        }
+        Utils.showDialog('Duplicate Printer', 'Duplicate', [{
+          type: "text",
+          label: "New Name",
+          property: "name"
+        }, {
+          type: "float",
+          label: "Width (mm)",
+          property: "width",
+          value: items[0].width
+        }, {
+          type: "float",
+          label: "Height (mm)",
+          property: "height",
+          value: items[0].height
+        }], function(settings) {
+          if (!settings.name) {
+            Utils.showOkDialog('Create Printer', ['A printer needs a name.']);
```

Comment:
Can set `required: true` in the field specifications above instead. For width and height too